packages/astro/src/core/render/core.ts

```diff
@@ -112,13 +112,6 @@ export async function renderPage(mod: ComponentInstance, ctx: RenderContext, env
 		Object.assign(pageProps, { components: (mod as any).components });
 	}
 
-	// HACK: expose `Fragment` for all MDX components
-	if (typeof mod.default === 'function' && mod.default.name.startsWith('MDX')) {
-		Object.assign(pageProps, {
-			components: Object.assign((pageProps?.components as any) ?? {}, { Fragment }),
-		});
-	}
```

Comment:
With the changes to the MDX integration in this PR, this code is unnecessary (but harmless) so could be removed.

But, could be a breaking change for people who use fragments in MDX pages currently and update `astro` but _don’t_ update `@astrojs/mdx`. Should I revert and we can remove in an `astro` major?

Another option could be to release the MDX changes as a minor (`0.13.0`) and add a minimum supported `astro` version to `peerDependencies`.

Reply:
Yeah this seems technically a breaking change. +1 on reverting for now and adding a TODO to remove in v2